src/dist/replication/meta_server/greedy_load_balancer.cpp

```diff
@@ -945,5 +964,77 @@ void greedy_load_balancer::report(const dsn::replication::migration_list &list,
         _recent_balance_copy_secondary_count->add(counters[COPY_SEC_COUNT]);
     }
 }
+
+std::string
+greedy_load_balancer::remote_command_balancer_ignored_app_ids(const std::vector<std::string> &args)
+{
+    static const std::string invalid_arguments("invalid arguments");
+    if (args.empty()) {
+        return invalid_arguments;
+    }
+    if (args[0] == "set") {
+        return set_balancer_ignored_app_ids(args);
+    }
+    if (args[0] == "get") {
+        return get_balancer_ignored_app_ids();
+    }
+    if (args[0] == "clear") {
+        return clear_balancer_ignored_app_ids();
+    }
+    return invalid_arguments;
+}
+
+std::string greedy_load_balancer::set_balancer_ignored_app_ids(const std::vector<std::string> &args)
+{
+    static const std::string invalid_arguments("invalid arguments");
+    dsn::zauto_write_lock l(_balancer_ignored_apps_lock);
```

Comment:
Move this line before `_balancer_ignored_apps = std::move(app_list);` to reduce lock scope.